### DISCUSSION TITLE
Connection Error

### DISCUSSION THREAD

**Discussion by sfmran77** [mckaywrigley/chatbot-ui#1658] [Help]
Hello, 
I managed to setup the chatbot and configured with openAI KEY. However, I can't receive the feedback and got error as below in console, any idea what leads to this problem?
![Snip20240410_6](https://github.com/mckaywrigley/chatbot-ui/assets/155049686/62957ea1-2951-4093-944a-afc3534ba9c1)

Thanks!

Immersive Translate INFO: sync remote rules success, latest: 2024/4/10 16:35:34
hot-reloader-client.js:197 [Fast Refresh] rebuilding
hot-reloader-client.js:243  GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.
processMessage @ hot-reloader-client.js:243
handler @ hot-reloader-client.js:395
fetch-models.ts:57 
        
        
        GET http://localhost:11434/api/tags net::ERR_CONNECTION_REFUSED
fetchOllamaModels @ fetch-models.ts:57
eval @ global-state.tsx:148
await in eval(异步)
eval @ global-state.tsx:152
commitHookEffectListMount @ react-dom.development.js:21035
commitHookPassiveMountEffects @ react-dom.development.js:23088
commitPassiveMountOnFiber @ react-dom.development.js:23193
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23201
commitPassiveMountEffects @ react-dom.development.js:23159
flushPassiveEffectsImpl @ react-dom.development.js:26372
flushPassiveEffects @ react-dom.development.js:26313
eval @ react-dom.development.js:26050
workLoop @ scheduler.development.js:256
flushWork @ scheduler.development.js:225
performWorkUntilDeadline @ scheduler.development.js:534
fetch-models.ts:78  Error fetching Ollama models: TypeError: Failed to fetch
fetchOllamaModels @ fetch-models.ts:78
await in fetchOllamaModels(异步)
eval @ global-state.tsx:148
await in eval(异步)
eval @ global-state.tsx:152
commitHookEffectListMount @ react-dom.development.js:21035
commitHookPassiveMountEffects @ react-dom.development.js:23088
commitPassiveMountOnFiber @ react-dom.development.js:23193
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23190
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23304
recursivelyTraversePassiveMountEffects @ react-dom.development.js:23171
commitPassiveMountOnFiber @ react-dom.development.js:23201
commitPassiveMountEffects @ react-dom.development.js:23159
flushPassiveEffectsImpl @ react-dom.development.js:26372
flushPassiveEffects @ react-dom.development.js:26313
eval @ react-dom.development.js:26050
workLoop @ scheduler.development.js:256
flushWork @ scheduler.development.js:225
performWorkUntilDeadline @ scheduler.development.js:534
fetch-models.ts:57 
        
        
        GET http://localhost:11434/api/tags net::ERR_CONNECTION_REFUSED
fetchOllamaModels @ fetch-models.ts:57
eval @ global-state.tsx:148
await in eval(异步)
eval @ global-state.tsx:152
commitHookEffectListMount @ react-dom.development.js:21035
invokePassiveEffectMountInDEV @ react-dom.development.js:23914
invokeEffectsInDev @ react-dom.development.js:26716
legacyCommitDoubleInvokeEffectsInDEV @ react-dom.development.js:26699
commitDoubleInvokeEffectsInDEV @ react-dom.development.js:26680
flushPassiveEffectsImpl @ react-dom.development.js:26389
flushPassiveEffects @ react-dom.development.js:26313
eval @ react-dom.development.js:26050
workLoop @ scheduler.development.js:256
flushWork @ scheduler.development.js:225
performWorkUntilDeadline @ scheduler.development.js:534
fetch-models.ts:78  Error fetching Ollama models: TypeError: Failed to fetch
fetchOllamaModels @ fetch-models.ts:78
await in fetchOllamaModels(异步)
eval @ global-state.tsx:148
await in eval(异步)
eval @ global-state.tsx:152
commitHookEffectListMount @ react-dom.development.js:21035
invokePassiveEffectMountInDEV @ react-dom.development.js:23914
invokeEffectsInDev @ react-dom.development.js:26716
legacyCommitDoubleInvokeEffectsInDEV @ react-dom.development.js:26699
commitDoubleInvokeEffectsInDEV @ react-dom.development.js:26680
flushPassiveEffectsImpl @ react-dom.development.js:26389
flushPassiveEffects @ react-dom.development.js:26313
eval @ react-dom.development.js:26050
workLoop @ scheduler.development.js:256
flushWork @ scheduler.development.js:225
performWorkUntilDeadline @ scheduler.development.js:534
index.ts:260 
        
       POST http://localhost:3000/api/chat/openai 500 (Internal Server Error)

**Comment by ghowe** [comment]
The connection errors are related to your Ollama configuration. Ensure that Ollama is installed and configured on the correct port.
You can change the host and port in the .env file:
# Ollama
NEXT_PUBLIC_OLLAMA_URL=http://localhost:11434

If Ollama is configured make sure that the port is not reserved or blocked. In windows you can check by issue the following command:
netsh int ipv4 show excludedportrange protocol=tcp

**Reply by sfmran77** [reply]
Thank you for the reply. I'm afraid this issue may not related to Ollama I suppose, since I'm trying to use OpenAI API but a local model. I'm not sure why this message appears.

Anyway, I tried the steps above also(install Ollama, check config and ports), and the problem is still there. Any new ideas will be appreciated.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/155049686/fcc5ae6a-3dbf-45f6-bb34-ebc51f0c3166)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/155049686/ef67059e-ab4e-4be7-a1a8-5589d5a9d746)

**Reply by sfmran77** [reply]
ah, and I'm using Mac, by the way

**Comment by Jonneal3** [comment]
@sfmran77 @ghowe you should join the discord --  https://discord.gg/K5hkY333kz
